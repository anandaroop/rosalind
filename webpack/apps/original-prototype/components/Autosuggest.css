.FairAutosuggest {
  margin: 0;
}

.FairAutosuggest ul.react-autosuggest__suggestions-list {
  list-style-type: none;
  margin: 1px 0 0 0;
  padding: 0;
  /*height: 300px;*/
  /*background: red;*/
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.FairAutosuggest li.react-autosuggest__suggestion {
  background: white;
  /*flex-basis: 50px;*/
  margin: 0 0 1px 0;
}

.FairAutosuggest li.react-autosuggest__suggestion--focused {
  background: purple;
  color: white;
}

.FairAutosuggest .FairSuggestion {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.FairAutosuggest .FairSuggestion {
  min-height: 2em;
}

.FairAutosuggest .FairSuggestion-name {
  flex: 1;
}
