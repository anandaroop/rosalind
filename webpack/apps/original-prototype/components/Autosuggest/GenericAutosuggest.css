ul.react-autosuggest__suggestions-list {
  /*list-style-type: none;*/
  /*margin: 1px 0 0 0;*/
  /*padding: 0;*/
  /*display: flex;*/
  /*flex-direction: column;*/
  /*justify-content: space-between;*/
}


li.react-autosuggest__suggestion {
  /*background: white;*/
  /*flex-basis: 50px;*/
  /*margin: 0 0 1px 0;*/
}

li.react-autosuggest__suggestion--focused {
  background: purple;
  color: white;
}
