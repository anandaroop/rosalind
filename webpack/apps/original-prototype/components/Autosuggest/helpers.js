import React from 'react'

export const getGeneSuggestionValue = (gene) => gene.name
export const renderGeneSuggestion = (gene) => <div>{gene.name}</div>

export const getTagSuggestionValue = (tag) => tag.name
export const renderTagSuggestion = (tag) => <div>{tag.name}</div>
