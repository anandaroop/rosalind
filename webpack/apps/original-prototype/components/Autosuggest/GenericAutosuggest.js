import React from 'react'
import Autosuggest from 'react-autosuggest'
import './GenericAutosuggest.css'

export default class GenericAutosuggest extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      value: '',
      suggestions: []
    }
    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this)
    this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this)
    this.onChange = this.onChange.bind(this)
    this.onSuggestionSelected = this.onSuggestionSelected.bind(this)
  }

  onSuggestionsFetchRequested ({value}) {
    this.props.fetchSuggestions(value).then(suggestions => {
      this.setState({ suggestions })
    })
  }

  onSuggestionsClearRequested () {
    this.setState({
      suggestions: []
    })
  }

  onChange (event, { newValue }) {
    this.setState({
      value: newValue
    })
  }

  onSuggestionSelected (event, { suggestion }) {
    this.props.selectSuggestion(suggestion)
    this.setState({value: ''})
  }

  render () {
    const inputElementProps = {
      placeholder: this.props.placeholder,
      value: this.state.value,
      onChange: this.onChange
    }

    return (
      <div className='GenericAutosuggest'>
        <Autosuggest
          id={this.props.id}
          suggestions={this.state.suggestions}
          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
          getSuggestionValue={this.props.getSuggestionValue}
          renderSuggestion={this.props.renderSuggestion}
          inputProps={inputElementProps}
          onSuggestionSelected={this.onSuggestionSelected}
          />
      </div>
    )
  }
}

// validation

const functionValidator = (signatureDescription) => {
  return (props, propName, componentName) => {
    const prop = props[propName]
    if ((prop instanceof Function) === false) {
      return new Error(
        `Invalid '${propName}' prop supplied to ${componentName}\n\t` +
        `'${propName}' should be a function with the signature (${signatureDescription})\n`
      )
    }
  }
}

GenericAutosuggest.propTypes = {
  id: React.PropTypes.string,
  placeholder: React.PropTypes.string,
  fetchSuggestions: functionValidator('searchTerm => listOfMatchingSuggestionObjects'),
  getSuggestionValue: functionValidator('suggestionObject => displayName'),
  renderSuggestion: functionValidator('suggestionObject => stringOrMarkupForSuggestionList'),
  selectSuggestion: functionValidator('suggestionObject => { handlerFunction(suggestionObject) }')
}
