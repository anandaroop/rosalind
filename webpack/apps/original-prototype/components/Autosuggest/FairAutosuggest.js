import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchFairs } from 'lib/rosalind-api'
import { getFairSuggestionValue, renderFairSuggestion } from './helpers'

function FairAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='fair-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchFairs}
      getSuggestionValue={getFairSuggestionValue}
      renderSuggestion={renderFairSuggestion}
      selectSuggestion={fair => { props.onSelectFair(fair) }}
      />
  )
}

FairAutosuggest.propTypes = {
  onSelectFair: React.PropTypes.func.isRequired
}

FairAutosuggest.defaultProps = {
  placeholder: 'Select a fair'
}

export { FairAutosuggest }
