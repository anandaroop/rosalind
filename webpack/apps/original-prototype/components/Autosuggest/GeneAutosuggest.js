import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes } from 'lib/rosalind-api'
import { getGeneSuggestionValue, renderGeneSuggestion } from './helpers'

function GeneAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='gene-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchGenes}
      getSuggestionValue={getGeneSuggestionValue}
      renderSuggestion={renderGeneSuggestion}
      selectSuggestion={gene => { props.onSelectGene(gene) }}
      />
  )
}

GeneAutosuggest.propTypes = {
  onSelectGene: React.PropTypes.func.isRequired
}

GeneAutosuggest.defaultProps = {
  placeholder: 'Select a gene'
}

export { GeneAutosuggest }
