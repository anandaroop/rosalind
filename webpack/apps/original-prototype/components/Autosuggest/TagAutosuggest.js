import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchTags } from 'lib/rosalind-api'
import { getTagSuggestionValue, renderTagSuggestion } from './helpers'

function TagAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='tag-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchTags}
      getSuggestionValue={getTagSuggestionValue}
      renderSuggestion={renderTagSuggestion}
      selectSuggestion={tag => { props.onSelectTag(tag) }}
      />
  )
}

TagAutosuggest.propTypes = {
  onSelectTag: React.PropTypes.func.isRequired
}

TagAutosuggest.defaultProps = {
  placeholder: 'Select a tag'
}

export { TagAutosuggest }
