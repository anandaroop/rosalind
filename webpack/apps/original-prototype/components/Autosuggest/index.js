import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes, matchTags, matchPartners } from 'lib/rosalind-api'
import { getGeneSuggestionValue, renderGeneSuggestion, getTagSuggestionValue, renderTagSuggestion, getPartnerSuggestionValue, renderPartnerSuggestion } from './helpers'

function GeneAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='gene-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchGenes}
      getSuggestionValue={getGeneSuggestionValue}
      renderSuggestion={renderGeneSuggestion}
      selectSuggestion={gene => { props.onSelectGene(gene) }}
      />
  )
}

GeneAutosuggest.propTypes = {
  onSelectGene: React.PropTypes.func.isRequired
}

GeneAutosuggest.defaultProps = {
  placeholder: 'Select a gene'
}

function TagAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='tag-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchTags}
      getSuggestionValue={getTagSuggestionValue}
      renderSuggestion={renderTagSuggestion}
      selectSuggestion={tag => { props.onSelectTag(tag) }}
      />
  )
}

TagAutosuggest.propTypes = {
  onSelectTag: React.PropTypes.func.isRequired
}

TagAutosuggest.defaultProps = {
  placeholder: 'Select a tag'
}

function PartnerAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='partner-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchPartners}
      getSuggestionValue={getPartnerSuggestionValue}
      renderSuggestion={renderPartnerSuggestion}
      selectSuggestion={partner => { props.onSelectPartner(partner) }}
      />
  )
}

PartnerAutosuggest.propTypes = {
  onSelectPartner: React.PropTypes.func.isRequired
}

PartnerAutosuggest.defaultProps = {
  placeholder: 'Select a partner'
}

export { GeneAutosuggest, TagAutosuggest, PartnerAutosuggest }
