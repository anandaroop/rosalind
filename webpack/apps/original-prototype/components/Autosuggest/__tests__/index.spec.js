import React from 'react'
import renderer from 'react-test-renderer'
import { GeneAutosuggest, TagAutosuggest, PartnerAutosuggest, FairAutosuggest } from '../../Autosuggest'

const mockHandler = jest.fn()

describe('GeneAutosuggest', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<GeneAutosuggest onSelectGene={mockHandler} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('TagAutosuggest', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<TagAutosuggest onSelectTag={mockHandler} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('PartnerAutosuggest', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<PartnerAutosuggest onSelectPartner={mockHandler} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('FairAutosuggest', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<FairAutosuggest onSelectFair={mockHandler} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
