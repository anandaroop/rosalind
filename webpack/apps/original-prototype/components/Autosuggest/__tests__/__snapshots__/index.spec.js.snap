exports[`GeneAutosuggest renders correctly 1`] = `
<div
  className="GenericAutosuggest">
  <div
    className="react-autosuggest__container">
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-gene-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a gene"
      role="combobox"
      type="text"
      value="" />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-gene-autosuggest" />
  </div>
</div>
`;

exports[`PartnerAutosuggest renders correctly 1`] = `
<div
  className="GenericAutosuggest">
  <div
    className="react-autosuggest__container">
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-partner-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a partner"
      role="combobox"
      type="text"
      value="" />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-partner-autosuggest" />
  </div>
</div>
`;

exports[`TagAutosuggest renders correctly 1`] = `
<div
  className="GenericAutosuggest">
  <div
    className="react-autosuggest__container">
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-tag-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a tag"
      role="combobox"
      type="text"
      value="" />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-tag-autosuggest" />
  </div>
</div>
`;
